src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -284,7 +287,7 @@ public int Available
                 {
                     // Update the state if we've become connected after a non-blocking connect.
                     _isConnected = true;
-                    _rightEndPoint = _nonBlockingConnectRightEndPoint;
+                    _rightEndPoint ??= _nonBlockingConnectRightEndPoint;
```

Comment:
In what situation will it be non-null?  I'm wondering in what situation we previously would have overwritten it and now we won't...

Reply:
I didn't think about this much. I replaced everything with a conditional assignment except for the one in `Bind`.

It can be non-null if for example you `Bind` first, and then do a `Connect` and then call `LocalEndPoint` and end up in this branch.
We don't want to overwrite it because that will lose the tracking (via `_rightEndPoint`) of the unix socket file we need to delete.

Reply:
Maybe it would make sense to add a test for this case, so we don't break it in the future.